GithubIssueTagger/InsertionChangelogGenerator.cs

```diff
@@ -0,0 +1,212 @@
+{
+    public class InsertionChangelogGenerator
+    {
+        public static async Task GenerateInsertionChangelog(GitHubClient gitHubClient)
+        {
+            // Adjust the sha/repo
+            var startSha = "58795a41a3c8c9d63801fb5072a2e68e48d7507e";
+            var orgName = "nuget";
+            var repoName = "nuget.client";
+            var branchName = "dev";
+            string[] issueRepositories = new string[] { "NuGet/Home", "NuGet/Client.Engineering" };
+            var resultHtmlPath = Path.Combine(Path.GetDirectoryName(typeof(InsertionChangelogGenerator).Assembly.Location), "results.html");
+
+            var githubBranch = await gitHubClient.Repository.Branch.Get(orgName, repoName, branchName);
+            var githubCommits = (await gitHubClient.Repository.Commit.Compare(orgName, repoName, startSha, githubBranch.Commit.Sha)).Commits.Reverse();
+            List<Commit> commits = await GetCommitDetails(gitHubClient, orgName, repoName, issueRepositories, githubCommits);
+
+            SaveAsHtml(commits, resultHtmlPath);
+        }
+
+        private static async Task<List<Commit>> GetCommitDetails(GitHubClient gitHubClient, string orgName, string repoName, string[] issueRepositories, IEnumerable<GitHubCommit> githubCommits)
+        {
+            var processedCommits = new List<Commit>();
+            foreach (var ghCommit in githubCommits)
+            {
+                var commit = new Commit(ghCommit.Sha, ghCommit.Author.Login, $"https://github.com/{orgName}/{repoName}/commit/{ghCommit.Sha}", ghCommit.Commit.Message);
+                var id = GetPRId(ghCommit.Commit.Message);
+                string pullRequestbody = string.Empty;
+                if (id != -1)
+                {
+                    commit.PR = new Tuple<int, string>(id, @"https://github.com/nuget/nuget.client/pull/" + id);
+                    pullRequestbody = (await gitHubClient.Repository.PullRequest.Get(orgName, repoName, id)).Body;
+                }
+
+                UpdateCommitIssuesFromText(commit, pullRequestbody, issueRepositories);
+                processedCommits.Add(commit);
+            }
+
+            return processedCommits;
+
+            static int GetPRId(string message)
+            {
+                foreach (Match match in new Regex(@"(#\d+)", RegexOptions.IgnoreCase).Matches(message))
+                {
+                    var pullRequestIdText = match.Value.Substring(1, match.Length - 1);
+                    int.TryParse(pullRequestIdText, out int prId);
+                    return prId;
+                }
+
+                return -1;
+            }
+        }
+
+        public static void UpdateCommitIssuesFromText(
+            Commit commit,
+            string body,
+            IList<string> issueRepositories)
+        {
+            var issueUrls = issueRepositories.SelectMany(e => GetIssueLinks(body, e)).ToList();
+            issueUrls.Remove("https://github.com/nuget/home/issues/1000");
```

Comment:
????

Reply:
This has been brought up in the past 2-3 retros, and several team meetings. It's in the issue template.

Reply:
wait what? How did I entirely miss us talking about a single, specific issues 3 times in a row, and what issue template is it in?

Reply:
It's in the PR template: https://raw.githubusercontent.com/NuGet/NuGet.Client/dev/.github/PULL_REQUEST_TEMPLATE.md. 

There were instances where the issue link was not added in a way GH could detect and that led to issues not getting closed.

Reply:
would it help if we change the template to use `issues/xxxx`, rather than `issues/1000`?

Reply:
> would it help if we change the template to use `issues/xxxx`, rather than `issues/1000`?

Not really. 
The tool looks for the url.

Reply:
a regex could help not match URLs that don't end in digits.

Reply:
You are free to provide a suggestion :)
But that'd require a template change, so the current tool wouldn't be good for those insertions.